packages/commerce-sdk-react/package.json

```diff
@@ -49,13 +49,15 @@
     "@tanstack/react-query": "^4.28.0",
     "@testing-library/jest-dom": "^5.16.5",
     "@testing-library/react": "^14.0.0",
-    "@types/js-cookie": "^3.0.3",
-    "@types/jsonwebtoken": "^9.0.0",
-    "@types/jwt-decode": "^3.1.0",
-    "@types/node": "^14.18.40",
-    "@types/react": "^18.2.0",
-    "@types/react-dom": "^18.2.1",
-    "@types/react-helmet": "^6.1.6",
+    "@types/js-cookie": "~3.0.3",
+    "@types/jsonwebtoken": "~9.0.0",
+    "@types/jwt-decode": "~3.1.0",
+    "@types/node": "~16.0.3",
+    "@types/prop-types": "~15.7.9",
+    "@types/react": "~18.2.0",
+    "@types/react-dom": "~18.2.1",
+    "@types/react-helmet": "~6.1.6",
+    "@types/react-router-dom": "~5.3.3",
```

Comment:
The versioning system for `@types` packages = major/minor are equal to the npm package they are defining, patch is used for bugfixes to the `@types` package. Therefore, when specifying `@types` dependencies, we should pin the major _and_ minor version (`~`), rather than just the major (`^`).

If we just pin the major, we could accidentally end up with `package@1.0.0` and `@types/package@1.2.3`, meaning that the types think we have access to features (introduced between `1.0.1` and `1.2.3`) that we can't actually use.

Reply:
I have a gut feeling that the scenario you defined is something that will actually happen unless we add the type package to the project far after we added the functional package (which is probably the case) and no rebuild the lock files. 

So the only time you'll have the scenario you defined if if we use package ^1.2.3 and lock the package lock, then X months later we decide to use add the types for said package by using "lerna add package@^1.2.3" which actually downloads 1.3.0 without disrupting the functional package version which is still locked to 1.2.3.


So my question is, if we left in the ^ for the types.. could we just have rebuilt the lock files? I guess there is also the issue of new versions of packages coming out and the types not being released at the same time 🤷

Reply:
`@types` packages are community maintained definitions for JS-only packages. Which means that there is no guarantee that any given version of any given package has a corresponding `@types` package, and new `@types` packages are all but guaranteed to not appear until after the main package is released. So we have to be fairly precise with our semver ranges (and sometimes even downgrade a bit, if we're on `package@1.2.3` but the latest `@types` is only `1.1.1` or whatever).